offer-manager-app/src/main/groovy/life/qbic/portal/offermanager/dataresources/persons/CustomerDbConnector.groovy

```diff
@@ -353,37 +354,35 @@ class CustomerDbConnector implements CreateCustomerDataSource, UpdateCustomerDat
   
   /**
    * @inheritDoc
-   * @param customerId
-   * @param updatedCustomer
    */
   @Override
-  void updateCustomer(String customerId, Customer updatedCustomer) {
+  void updateCustomer(int oldCustomerId, Customer updatedCustomer) {
 
-        int oldCustomerId = Integer.parseInt(customerId)
-      if (getCustomer(oldCustomerId)==null) {
-        throw new DatabaseQueryException("Customer is not in the database and can't be updated.")
-      }
+    if (!getCustomer(oldCustomerId)) {
+      throw new DatabaseQueryException("Customer is not in the database and can't be updated.")
+    }
             
-      Connection connection = connectionProvider.connect()
-      connection.setAutoCommit(false)
+    Connection connection = connectionProvider.connect()
+    connection.setAutoCommit(false)
 
-      connection.withCloseable {it ->
-        try {
-          int newCustomerId = createNewCustomer(it, updatedCustomer)
-          storeAffiliation(it, newCustomerId, updatedCustomer.affiliations)
-          connection.commit()
-          
-          // if our update is successful we set the old customer inactive
-          changeCustomerActiveFlag(oldCustomerId, false)
+    connection.withCloseable {it ->
+      try {
+        int newCustomerId = createNewCustomer(it, updatedCustomer)
+        //todo also associate the affiliations of the old entry to the customer
+        println updatedCustomer.affiliations
+        storeAffiliation(it, newCustomerId, updatedCustomer.affiliations)
```

Comment:
```suggestion
        List<Affiliation> allAffiliations = fetchAffiliationsForPerson(oldCustomerId)
        allAffiliations.addAll(updatedCustomer.affiliations)
        storeAffiliation(it, newCustomerId, allAffiliations)
```
should work like this? needs to be tested though

Reply:
If we want to be save we could put it before the           changeCustomerActiveFlag(oldCustomerId, false)

Reply:
nevermind, you already did that 👍

Reply:
_Runs on my machine_™